plugins/autoid/plugin.js

```diff
@@ -1,9 +1,95 @@
 'use strict';
 
 (function () {
-	CKEDITOR.plugins.add('autoid', {
-		init: function (editor) {
+  var editorHasFocus = false;
+  var commandIsActive = false;
+  var EVENT_NAMES = {
+    ALL_IDS_COMPLETE: 'allIdsComplete',
+    ID_ADDED: 'idAdded'
+  };
 
-		}
-	});
+  CKEDITOR.plugins.add('autoid', {
+    init: function (editor) {
+      var self = this;
+
+      editor.addFeature({
+        allowedContent: 'h1 h2 h3 h4 h5 h6 *[id]'
+      });
+
+      if (editor && !editor.config.autoid) {
+        editor.config.autoid = {};
+      }
+      self.settings = editor.config.autoid;
+      if (!self.settings) {
+        self.settings = {};
+      }
+
+      editor.addCommand('autoid', {
+        exec: function (editor) {
+          if (!commandIsActive) {
+            start();
+          } else {
+            stop();
+          }
+        },
+        editorFocus: true
+      });
+
+      editor.on("instanceReady", function () {
+        if (self.settings.autostart !== false) {
```

Comment:
isn't `undefined` falsy in JS?   I just did `console.log(!!self.settings.autostart)` and it returned `false`